app/src/main/java/org/mozilla/fenix/settings/creditcards/String.kt

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.creditcards
+
+private const val MAX_CREDIT_CARD_NUMBER_LENGTH = 19
+private const val MIN_CREDIT_CARD_NUMBER_LENGTH = 12
+
+/**
+ * Strips characters other than digits from a string.
+ * Used to strip a credit card number user input of spaces and separators.
+ */
+
+fun String.toCreditCardNumber(): String {
+    return this.filter { it.isDigit() }
+}
+
+/**
+ * Uses string size and Luhn Algorithm check (https://en.wikipedia.org/wiki/Luhn_algorithm)
+ * to validate a credit card number
+ */
+@Suppress("MagicNumber")
+fun String.validateCreditCardNumber(): Boolean {
+    val creditCardNumber = this.toCreditCardNumber()
+
+    if (creditCardNumber != this) return false
+
+    // credit card numbers have at least 12 digits and at most 19 digits
+    if (creditCardNumber.length < MIN_CREDIT_CARD_NUMBER_LENGTH ||
+        creditCardNumber.length > MAX_CREDIT_CARD_NUMBER_LENGTH
+    ) return false
+
+    // Luhn Algorithm check
```

Comment:
We can probably remove this comment since the function name should be self explanatory now